### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by vks**
This reduces the number of non-crates.io dependencies to 1.

### CODE REVIEW

**Comment by michaelkirk** [comment]
Nice! I see a couple new duped dependencies in Cargo.lock - any idea what introduced them?

**Comment by dabreegster** [comment]
> Nice! I see a couple new duped dependencies in Cargo.lock - any idea what introduced them?

I glanced at a few...
- `libloading` because `glutin` is still using an older version
- `memmap2` because seemingly `fontdb` and `usvg` use different versions. Same author! I'll send a PR for that
- Haven't detangled `darling` yet...

**Review by dabreegster — APPROVED** [review]
Thanks, overall LGTM! More red than green in `Cargo.lock` is always nice. We like to watch out for multiple versions of the same dependency creeping in to keep builds fast(er) and small(er). Especially in the geo ecosystem, this winds up taking some grunt work to update everything when the core libraries change.

**Comment by vks** [comment]
My main goal was to move back to upstream where possible, I did not really consider duplicate dependencies to be honest.

If reducing build times / sizes is a goal, it may be worth it to get rid of heavy dependencies such as reqwest.

**Comment by dabreegster** [comment]
> My main goal was to move back to upstream where possible, I did not really consider duplicate dependencies to be honest.

I think that matches our priorities as well. We just like to also keep an eye on the duplication, because we've invested time in trimming it down previously. Most of that effort was in the geo world, IIRC.

I'm fine merging this unless @michaelkirk has any concerns

> it may be worth it to get rid of heavy dependencies such as reqwest.

That'd be nice. `abstio/src/download.rs` tries to wrap it and add some interface for plumbing back progress to both a CLI caller (in the updater) and a UI (in `map_gui`). I haven't explored the lighter-weight alternatives yet.

File: widgetry/Cargo.toml [hunk 1]
```diff
@@ -15,19 +15,19 @@ anyhow = "1.0.38"
...
-glow = "0.8.1"
+glow = "0.9.0"
 glutin = { version = "0.26.0", optional = true }
 htmlescape = "0.3.1"
 image = { version = "0.23.12", default-features = false, features=["png"] }
 instant = "0.1.7"
 log = "0.4.14"
 lru = "0.6.4"
-lyon = "0.16.2"
+lyon = "0.17.0"
```
[the remark this answers is not in the document]
Comment (dabreegster) [reply widgetry/Cargo.toml:25]: I spoke too soon. If you try to click on a lane, I get a crash:
`assertion failed: !self.in_subpath', /home/dabreegster/.cargo/registry/src/github.com-1ecc6299db9ec823/lyon_path-0.17.2/src/private.rs:55:13`
Let's revert this for now and file a bug later.

**Review by dabreegster — CHANGES_REQUESTED** [review]
`lyon` has a regression

**Comment by michaelkirk** [comment]
> I think that matches our priorities as well. We just like to also keep an eye on the duplication, because we've invested time in trimming it down previously. Most of that effort was in the geo world, IIRC.
>
> I'm fine merging this unless @michaelkirk has any concerns

I agree it's a balance of a lot of priorities — Up to date is good, fast builds are good. 

In particular with the geo crates, I'd prefer staying off master a little longer rather than duplicating. For one, I'm familiar with the releases and know we're not missing anything that affects us, and also, honestly, keeping abstreet's geo deps in sync is often the motivation that the downstream geo crates get their geo dep updated. 😅 

So, I'd prefer to not merge anything that dupes our geo dependencies unless we have a specific reason that we can't update a downstream crate. If you leave those out for now, I can follow up with getting them all updated.

**Comment by dabreegster** [comment]
I switched to git for https://github.com/urschrei/polylabel-rs, since dependabot has been faithfully keeping up there, but crates.io still needs a release. Now we're not duplicating any geo dependencies.

I also rolled back lyon to the previous version to avoid that crash.

@vks, thanks for doing this maintenance! Merging now.

**Pull request merged**